src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -835,6 +845,41 @@ pub enum RoundMode {
 }
 
 impl Decimal {
+    /// abs the Decimal into a new Decimal.
+    #[inline]
+    pub fn abs(mut self) -> Res<Decimal> {
+        self.negative = false;
+        Res::Ok(self)
+    }
+
+    /// ceil the Decimal into a new Decimal.
+    pub fn ceil(&self) -> Res<Decimal> {
+        let mut target = if !self.frac_part_is_zero() && !self.negative {
+            let dec1 = Decimal::from(1i64);
+            self + &dec1
+        } else {
+            Res::Ok(self.clone())
+        };
+        if let Res::Ok(t) = target {
+            target = t.round(0, RoundMode::Truncate);
+        }
+        target
+    }
+
+    /// floor the Decimal into a new Decimal.
+    pub fn floor(&self) -> Res<Decimal> {
+        let mut target = if !self.frac_part_is_zero() && self.negative {
```

Comment:
How about use `t.round(0, RoundMode::Truncate)` when it's positive otherwise use `t.round(0, RoundMode::Ceiling)`?